pkg/apis/softwarecomposition/network_types.go

```diff
@@ -0,0 +1,65 @@
+package softwarecomposition
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Protocol string
+type CommunicationType string
+
+const (
+	ProtocolTCP  Protocol = "TCP"
+	ProtocolUDP  Protocol = "UDP"
+	ProtocolSCTP Protocol = "SCTP"
+
+	CommunicationTypeIngress CommunicationType = "internal"
+	CommunicationTypeEgress  CommunicationType = "external"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkNeighborsList is a list of NetworkNeighbors.
+type NetworkNeighborsList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+
+	Items []NetworkNeighbors
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkNeighbors represents a list of network communications for a specific workload.
+type NetworkNeighbors struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec NetworkNeighborsSpec
+}
+
+type NetworkNeighborsSpec struct {
+	metav1.LabelSelector // The labels which are inside spec.selector in the parent workload.
+	Ingress              []NetworkEntry
+	Egress               []NetworkEntry
+}
+
+// IngressEntry represents a single network communication.
+type NetworkEntry struct {
```

Comment:
chore: we must make the naming match across comments and type definitions

Also, I have trouble understanding the comment. What’s a “single network communication”? And Network Entry seems too generic. It seems that we are describing a single network neighbor.

If we call our top-level resources Network Neighbors, wouldn’t it make sense that the entries in the `Spec.Ingress` list would have the type `NetworkNeighbor` singular? It seems to be less confusing, in my opinion.

If you do change the name, please keep in mind you have to change them in both top-level and `v1beta1` directories and also re-generate the code.

Reply:
suggestion: how about we call this `NetworkNeighbor`, singular?

I think this will be more immediately obvious than calling it a generic `NetworkEntry`.